### DISCUSSION TITLE
Minimal and fast model configuration

### DISCUSSION THREAD

**Discussion by aiwalter** [ourownstory/neural_prophet#222] [Q&A - get help using NeuralProphet] [answered]
I am running some tests with `neuralprophet` and I need to find a minimal and fast model configuration to speed it up a bit. Any ideas for this configuration?

**Comment by aiwalter** [comment]
My current model is this, which is now much faster already compared to a model with only default attributes.
```python
m = NeuralProphet(epochs=1, batch_size=1, train_speed=3, learning_rate=100)
metrics = m.fit(df, freq="Y")
```

**Comment by ourownstory** [comment, marked as answer]
Ciao @aiwalter 

First, when trying to speed up training, don't compromise on validation accuracy too much.
`metrics = m.fit(df, validate_each_epoch=True, valid_p=0.2)`
(note: retrain without validation before making actual forecasts)

Further, `train_speed` modifies the default values of `epochs`, `batch_size` and `learning_rate`.
If you set them manually, do not set `train_speed`. Tip: a larger batch_size actually leads to faster training, usually. I would recommend first having a look at the default values selected and start with those.

Finally, you can read up [about the settings in the documentation](https://ourownstory.github.io/neural_prophet/hyperparameter-selection/)

Do yo mind sharing, what dataset and what computer you are working with?

**Reply by diegulio** [reply]
hi @ourownstory , what happens with` train_speed` parameter? seems to have been removed, it throws me an error when I adjust it ):

**Reply by ourownstory** [reply]
Hi @diegulio 
We decided to remove it because it was not clear enough what it actually does, and thus was often used in misleading ways.
What it used to do is to half/double the lr, batch-size and epochs for each step change in the parameter. 

If you like to speed up training (potentially at the cost of accuracy), you can observe the automatic batch-size, number of epochs and learning-rate and then fine tune those to your liking. The automatic ones are printed out as INFO log statements.

**Reply by diegulio** [reply]
I see! Thanks for the fast answer! :D
